core/src/rpc.c

```diff
@@ -20,7 +22,56 @@ static int check_version(InternalCommandRequest *cmd) {
   }
   return true;
 }
+/**
+ * Check signature on input bytes.
+ * Deserialize them and then populate the internal command request's
+ * redundant fields with the deserialized data.
+ * @return Return an error if the signature check failed or decoding the bytes failed.
+ */
+static Result populate_internal_command(InternalCommandRequest * to){
+  
+  Result res = Result_SUCCESS;
+  if(!check_signature(to->serialized_command_request.bytes, to->serialized_command_request.size, to->qrsignature.signature.bytes)){
+    res = Result_SIG_CHECK_FAILED;
+    ERROR("Signature check on the binary payload failed.");
+    goto cleanup;
+  
+  }
+  pb_istream_t pb_command = pb_istream_from_buffer(to->serialized_command_request.bytes, to->serialized_command_request.size);
+  CommandRequest from = CommandRequest_init_default;
+  if(!pb_decode(&pb_command, CommandRequest_fields, &from)){
```

Comment:
nit: add a comment on why we use here `pb_decode` and not `pb_decode_delimited` as elsewhere in this file?